new-frontiers-14/frontier-station-14
# Loadouts: Fallbacks as defaults, less redundancy, harpy/diona fixes

## About the PR



Uses fallback items as defaults to be selected as defaults for empty or invalid loadouts.  For completely empty loadouts, we take as many fallback items exist until we reach the maximum number of supported items per loadout group.  For existing loadouts, we assume the user intended to remove items, so we restore a minimal number of items.

Added subgroups to loadoutGroups to reduce redundancy.  Subgroups append their loadout entries to the menu in the Character Setup UI, checking the effects on each before being appended.  This solves the issues brought up in #1721 regarding the STC shoes needing to copy every entry from the Contractor shoes.

Added "hideEffects" LoadoutEffects to each loadout - these are used both for validation, and to hide entries from the loadout selection window.  A harpy character, for instance, shouldn't be presented with options for jumpsuits with pants when they can't ever wear them.  Diona also shouldn't be presented with the options for shoes.

Additionally, this PR adds loadout effects for harpy-compatible items to all non-skirt jumpsuits.  NOTE: certain items (monastic robes, kilt) should _probably_ be individually marked as HarpyClothing.  All shoes have also been marked as invalid for diona.

NOTES:
* Currently, the Pilot and Mercenary jobs receive _all_ available items for the previous job tiers.  This may be undesirable (what makes a pilot special?) and is a change from previous behaviour.

## Why / Balance


#1700 provided good motivation to provide medical insurance trackers for default characters, and the discussion motivated the possibility of an encryption key-less loadout.

## How to test


1. Open the character selection screen.
2. Create a new character.
3. Set their species to Harpy.  Their appearance should immediately swap into a skirt.
4. Open the Contractor loadout, swap to the Jumpsuit tab.  Only skirts should be listed.
5. Close the loadout, swap species.
6. Open the Contractor loadout, swap to the Jumpsuit tab, select a non-skirt jumpsuit.
7. Swap species to Harpy again.  Your character should swap back into a jumpskirt, but currently, this does not happen.
8. Save your character.  The outfit the paper doll is wearing should return to a jumpskirt.
9. Open the STC loadout selection.  Go to the Shoes tab.  Laceups should be available for free.  Scroll down, there should be no other entries for laceup shoes.

## Media


The in-game Contractor jumpsuit selection screen for a harpy.  Note how no jumpsuits with pants are available.
![image](https://github.com/user-attachments/assets/e5800310-9aab-4155-b632-97dec7210bb3)

The in-game Mercenary headgear selection screen for a harpy.  The pilot's helmet is being appended from the PilotHead subgroup, and everything below is being appended from the ContractorHead subgroup.
![image](https://github.com/user-attachments/assets/5485f265-929a-46ac-97b8-867c5b7d0da9)


## Breaking changes


**Changelog**



:cl:
- tweak: New characters now start with a medical tracker by default in their loadout.
- fix: Harpies cannot select jumpsuits without skirts in their loadout, Diona do not spawn with shoes.
